source/13_envs/evogym.rst

```diff
@@ -0,0 +1,226 @@
+Evogym
+~~~~~~~~~
+
+Overview
+=========
+
+Evolution Gym (Evogym) is the first large-scale benchmark for co-optimizing the design and control of soft robots.
+Each robot is composed of different types of voxels (e.g., soft, rigid, actuators), resulting in a modular and expressive robot design space.
+The environment spans a wide range of tasks, including locomotion on various types of terrains and manipulation.
+The image below shows a robot in action.
+
+.. image:: ./images/evogym.gif
+   :align: center
+
+Install
+========
+
+Installation Method
+--------------------
+
+The environment can be downloaded and installed using GitHub and pip.
+Given that the installation procedure may vary a little according to your OS, you are invited to check the instructions in the original `repository <https://github.com/EvolutionGym/evogym#installation>`__ in case you should encounter any problem.
+
+.. code:: shell
+
+   git clone --recurse-submodules https://github.com/EvolutionGym/evogym.git
+   cd evogym
+   pip install -r requirements.txt
+   python setup.py install
+
+Verify Installation
+--------------------
+
+After the installation is complete, you can verify that the installation was successful by running the following command on the command line:
+
+.. code:: shell
+
+   python gym_test.py
+
+Space Before Transformation (Original Environment)
+===================================================
+
+Observation Space
+------------------
+
+- The observation space includes the state information of the robot, a\ ``(2N + 3)``\ vector including the relative position of each voxel corner with respect to the center of mass of the robot \ ``(2N)``\, and the velocity and orientation of center of mass \ ``(3)``\. The data type is\ ``float32``\.
+- To handle complex tasks, specifically those with varying terrain types, an additional observation vector including terrain information is provided.
+- Furthermore, goal-related information is offered to inform the controller of the execution status of the current task.
+- For instance, in manipulation tasks where the robot interacts with some object O, it is provided orientation and velocity as well as the position of O’s center of mass relative to the robot.
+
+Action Space
+-------------
+
+- It is a a continuous action space of size N. Each of the N components of the action vector is associated with an actuator voxel (either horizontal or vertical) of the robot, and instructs a deformation target of that voxel. The data type is \ ``float``\.
```

Comment:
The action space is continuous of size N. Each component of the N-size action vector is associated with an actuator voxel (either horizontal or vertical) of the robot, and instructs a deformation target of that voxel, compressing or stretching the size of voxel from 60% to 160%。